### PR TITLE
Give RangeFinder backends ownership of their parameters

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_RangeFinder/RangeFinder.h [hunk 1]
```diff
@@ -32,6 +32,7 @@ class RangeFinder
 {
 public:
     friend class AP_RangeFinder_Backend;
+    friend class AP_RangeFinder_analog; // bodgy detect function
```
Comment (OXINARF) [inline libraries/AP_RangeFinder/RangeFinder.h]: Not needed? Looks like the *bodgy detect function* is fixed in the second commit.

File: libraries/AP_RangeFinder/RangeFinder.h [hunk 2]
```diff
@@ -190,12 +195,15 @@ class RangeFinder
...
+        return _RangeFinder_STATE(instance).pos_offset;
     }
     const Vector3f &get_pos_offset_orient(enum Rotation orientation) const;
 
-private:
+protected:
+
     RangeFinder_State state[RANGEFINDER_MAX_INSTANCES];
+
+private:
```
Comment (OXINARF) [inline libraries/AP_RangeFinder/RangeFinder.h]: Sorry, I forgot about this one earlier. Why is this change needed? I'm guessing it was also because of the *analog* detect method too?

**Comment by peterbarker** [comment]
On Mon, 7 Aug 2017, Francisco Ferreira wrote:

> +
> +private:
> 
> Sorry, I forgot about this one earlier. Why is this change needed? I'm
> guessing it was also because of the analog detect method too?

Yep, fixed.

**Comment by OXINARF** [comment]
Merged, thanks!
